src/ts-serialize.ts

```diff
@@ -11,13 +11,14 @@
 
 import * as util from 'util';
 
-import {Class, Document, Function, Interface, Method, Namespace, Param, Property} from './ts-ast';
+import * as ts from './ts-ast';
 
 /**
- * Encode a TypeScript AST node in TypeScript declaration file syntax (d.ts).
+ * Serialize a TypeScript declarations file AST to a string.
  */
 export function serializeTsDeclarations(
```

Comment:
What do you think about inverting this. Making the different type types each their own class and giving them a serialize() method

Reply:
I think I prefer keeping the AST defined as pure data objects, seems simpler to me.